crates/sargon/src/keys_collector/collector/derivation_purpose.rs

```diff
@@ -0,0 +1,91 @@
+use crate::prelude::*;
+
+/// The purpose that initiated an interaction with the host to derive keys.
+/// The orchestrator behind this operation is the `KeysCollector`.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum DerivationPurpose {
+    /// When the create account flow, initiates keys collection
+    /// for account VECIs
+    CreatingNewAccount,
+
+    /// When the create persona flow, initiates keys collection
+    /// for identity VECIs
+    CreatingNewPersona,
+
+    /// When applying a security shield to an account, initiates keys collection
+    /// for account MFA
+    SecurifyingAccount,
+
+    /// When applying a security shield to a persona, initiates keys collection
+    /// for identity MFA
+    SecurifyingPersona,
+
+    /// When adding a new factor source, initiates keys collection
+    /// for collecting various factor instances.
+    PreDerivingKeys,
+}
+
+impl DerivationPurpose {
+    pub fn new_for_creating(entity_kind: CAP26EntityKind) -> Self {
+        match entity_kind {
+            CAP26EntityKind::Account => Self::CreatingNewAccount,
+            CAP26EntityKind::Identity => Self::CreatingNewPersona,
+        }
+    }
+
+    pub fn new_for_securifying(entity_kind: CAP26EntityKind) -> Self {
```

Comment:
`new_for_securifying` -> `for_securifying_or_updating` (drop `new` and include `updating` ) I think

Reply:
We use this `new_*` for whenever we instantiate any type. How can someone know when new is suitable or not?

Reply:
Personally I would always go with a descriptive name like `for_securifying_or_updating`, but I tried to be consistent with the rest of the codebase.